### PR TITLE
Feature/AD-33 View Profiles

### PR DESCRIPTION
**Pull request by garethdandrews**
The application now features user profiles. There is a link in the navbar to the profile of `current_user`, and there are links to a project owners profile on the project card. The styling of the cards has been updated to include two clickable sections, the main content (title, summary) and the footer (owner, avatar, view count, likes etc ...).
The user profile is not yet editable, but will be included on a future branch. 
There are preliminary stylings on the user profile image, so in the future, the user can change their profile image by clicking the image.

Screenshot 2021-08-18 at 11 06 09

Screenshot 2021-08-18 at 11 06 21

Screenshot 2021-08-18 at 11 06 48

Hovering the different sections of the project card:
Screenshot 2021-08-18 at 11 07 25

Screenshot 2021-08-18 at 11 07 32

Mobile:
The links on mobile are underlined to indicate that they're clickable.

Screenshot 2021-08-18 at 11 08 52

Additionally on this branch I have tidied up some of the SCSS. This includes extracting some of the styles from forms and putting them into their own class called 'container'. This includes styles for box shadows, padding and rounded edges of the forms.

### CODE REVIEW

**Comment by phil-6** [comment]
The yellow on white contrast ratio looks to be quite low. Might be worth picking a colour other than yellow for that.

**Comment by garethdandrews** [comment]
> The yellow on white contrast ratio looks to be quite low. Might be worth picking a colour other than yellow for that.

How about a slightly darker yellow?
Screenshot 2021-08-18 at 11 27 44

File: app/assets/stylesheets/project_cards.scss [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        }
+
+        .description {
+          margin-top: 1rem;
+          margin-bottom: 0;
+          list-style: inside;
+          align-content: center;
+        }
+
+
```
Comment (phil-6) [inline app/assets/stylesheets/project_cards.scss:164]: extra empty line can be removed

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,14 @@
...
+      <div class="avatar">
+        <img class="image" src=<%= @user.avatar %> alt="avatar">
+      </div>
+      <h1><%= "#{@user.first_name} #{@user.last_name}" %></h1>
+      <p><%= @user.bio %></p>
+      <small>Member since <%= @user.created_date %> | Last online <%= @user.last_update_date%> </small>
+    </div>
+  </div>
+  <%= render 'projects/project_card_collection', projects: @user.projects.all %>
+</section>
```
Comment (phil-6) [inline app/views/users/show.html.erb:14]: Is it worth showing something to do with the number of comments the user has made?

Comment (garethdandrews) [reply app/views/users/show.html.erb:14]: So something like a recent activity view?

Comment (garethdandrews) [reply app/views/users/show.html.erb:14]: Or comment history?

Comment (phil-6) [reply app/views/users/show.html.erb:14]: Or just comment count?

Comment (garethdandrews) [reply app/views/users/show.html.erb:14]: can do 👍

**Comment by phil-6** [comment]
> > The yellow on white contrast ratio looks to be quite low. Might be worth picking a colour other than yellow for that.
> 
> How about a slightly darker yellow?

As much as I want to, I never like dark yellows. Maybe use the purple or pink?

**Review by AlexW555 — APPROVED** [review]
LGTM ✅

**Pull request merged**